libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/PrimaryRouteUpdateTask.java

```diff
@@ -1,66 +1,97 @@
 package com.mapbox.services.android.navigation.ui.v5.route;
 
-import android.os.AsyncTask;
+import android.os.Handler;
 
 import com.mapbox.geojson.Feature;
 import com.mapbox.geojson.FeatureCollection;
 
 import java.lang.ref.WeakReference;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import static com.mapbox.services.android.navigation.ui.v5.route.RouteConstants.FIRST_COLLECTION_INDEX;
 import static com.mapbox.services.android.navigation.ui.v5.route.RouteConstants.PRIMARY_ROUTE_PROPERTY_KEY;
 
-class PrimaryRouteUpdateTask extends AsyncTask<Void, Void, List<FeatureCollection>> {
+class PrimaryRouteUpdateTask extends Thread {
```

Comment:
```suggestion
class PrimaryRouteUpdateTask implement Runnable {
```

Reply:
We want to run the task in a separate thread i.e. with `Runnable` we'd need to run it explicitly in a different thread in the call-site (using a `ThreadPoolExecutor` or creating the `Thread` and passing the `Runnable`. That might need some refactoring that we can implement in a future / separate PR.